app/models/mixins/saved_url_mixin.rb

```diff
@@ -0,0 +1,15 @@
+module SavedUrlMixin
+  extend ActiveSupport::Concern
+
+  def saved_url=(url, userid)
+    # , :userid => userid).each(&:destroy)
+    SavedUrl.where(:refers_to_id => id, :refers_to_type => self.class.name).each(&:destroy)
+    save_url = SavedUrl.create!(
+      :refers_to_id   => id,
+      :refers_to_type => self.class.name,
+      #:userid       => User.find_by(:userid => userid),
+      :url          => url
+    )
+    save_url.id
```

Comment:
The `save_url` variable is not used, so no need to allocate it.

Again with polymorphic:
```ruby
SavedUrl.create!(:resource => self, :url => url).id
```

Or something like `create_saved_url!(:url => url).id`